web/blog/2024-08-13-how-to-add-auth-with-lucia-to-your-react-nextjs-app.md

```diff
@@ -0,0 +1,623 @@
+  return (
+    <div style={{ maxWidth: '400px', margin: '0 auto' }}>
+      <SignupForm />
+      <br />
+      <span>
+        I already have an account (<Link to="/login">go to login</Link>).
+      </span>
+    </div>
+  )
+}
+```
+
+And finally, for protecting routes, is as simple as changing it in `main.wasp` adding **`authRequired: true`** , so, we can simply add it like this:
+
+```tsx
+// main.wasp
+page MainPage {
+  component: import Main from "@src/pages/Main",
+  authRequired: true
+}
+```
+
+If you’d like to check this example in more depth, feel free to check this repo here: [wasp/examples/todo-typescript at release · wasp-lang/wasp (github.com)](https://github.com/wasp-lang/wasp/tree/release/examples/todo-typescript).
+Other great place to check is their documentation, which can be found [here](https://wasp-lang.dev/docs/auth/overview). It covers most of what I said here, and even more (e.g. the awesome new [hooks](https://wasp-lang.dev/docs/auth/auth-hooks) that came with Wasp v0.14)
+
+![https://media4.giphy.com/media/nDSlfqf0gn5g4/giphy.gif?cid=7941fdc6oxsddr7p8rjsuavcyq7ugiad8iqdu1ei25urcge4&ep=v1_gifs_search&rid=giphy.gif&ct=g](https://media4.giphy.com/media/nDSlfqf0gn5g4/giphy.gif?cid=7941fdc6oxsddr7p8rjsuavcyq7ugiad8iqdu1ei25urcge4&ep=v1_gifs_search&rid=giphy.gif&ct=g)
+
+Way easier, isn’t it? Let’s review the steps we took to get here:
+
+- Set up the project.
+- Add the User entity to the database.
+- Define authentication in the main Wasp configuration.
+- Obtain GitHub OAuth credentials and configure your environment variables.
+- Add routes and pages for login and signup with pre-built, easy-to-use components.
+- Protect routes by specifying `authRequired` in your configuration.
+
+### Customizing Wasp Auth
+
+If you need more control and customization over the authentication flow, Wasp provides Auth hooks that allow you to tailor the experience to your app's specific needs. These hooks enable you to execute custom code during various stages of the authentication process, ensuring that you can implement any required custom behavior.
+
+For more detailed information on using Auth hooks with Wasp, visit the [Wasp documentation](https://wasp-lang.dev/docs/auth/auth-hooks).
+
+### Bonus Section: Adding Email/Password Login with Wasp and Customizing Auth
+
+Now let’s imagine we want to add email and password authentication — with all the usual features we’d expect that would follow this login method (e.g. reset password, email verification, etc.). 
+
+With Wasp, all we have to do is add a few lines to your main.wasp file, so, simply updating your Wasp configuration to include email/password authentication makes it work straight out of the box!
+
+![https://wasp-lang.dev/img/auth-ui/auth-demo-compiler.gif](https://wasp-lang.dev/img/auth-ui/auth-demo-compiler.gif)
+
+Wasp will handle the rest, also updating UI components and ensuring a smooth and secure authentication flow.
+
+```tsx
+//main.wasp
```

Comment:
@mmaksimovic @vincanger I noticed that the Wasp file is not highlighted correctly since `tsx` is used and we are using a comment to provide a title.

We have support for highlighting Wasp files and there is way to provide a title:
```
```wasp title="main.wasp
// stuff
```
It goes from this:
![Screenshot 2024-08-14 at 12 27 41](https://github.com/user-attachments/assets/73f45412-a7bd-44b0-a391-0c296647298e)

To this:
![Screenshot 2024-08-14 at 12 29 06](https://github.com/user-attachments/assets/9f87d0ea-ea9c-42f2-b2a8-663060eb9746)